ts/Core/Renderer/SVG/SVGElement.ts

```diff
@@ -1612,13 +1614,23 @@ class SVGElement implements SVGElementLike {
         const x = Math.min(aX, bX, cX, dX),
             y = Math.min(aY, bY, cY, dY),
             boxWidth = Math.max(aX, bX, cX, dX) - x,
-            boxHeight = Math.max(aY, bY, cY, dY) - y;
+            boxHeight = Math.max(aY, bY, cY, dY) - y,
+            [xCoord, yCoord] = ['x', 'y'].map((str): number => {
+                const attr = this.attr(str);
+                return isNumber(attr) ? attr : 0;
+            });
```

Comment:
It shouldn't be necessary to adjust the coordinates like this. There seems to be something wrong with the corner coordinates in the first place.

Consider this demo (running on this PR branch, checked out locally, as of today): https://jsfiddle.net/highcharts/a2L30w9v/

Observe that
* For a `label`, the bounding box rectangle is wrong. For a `text`, the bounding box rectangle is correct. Which shows that your correction for the x and y position should be applied earlier, to the `aX`, `bX` etc. constants.
* The polygon is flipped.

Reply:
And to elaborate, note that the bounding box for labels is wrong in the master too: https://jsfiddle.net/highcharts/a2L30w9v/1/

We might consider that an issue separate from this PR, but in any case this PR depends on it. We won't get the polygons right unless we fix the bounding box. @MarkusBarstad if you see fit, you can create a new issue on https://jsfiddle.net/highcharts/a2L30w9v/1/ and work on that separately, that's up to you.

Reply:
PS: The bounding box problem for labels also apply without rotation: https://jsfiddle.net/highcharts/a2L30w9v/2/. So this is likely an old problem. I think it comes from the SVGLabel being positioned by setting translation to the `g` element. Changes may also require updates in the [overlapping functionality](https://github.com/highcharts/highcharts/blob/e7b36f31692df459748ac15ab25b66a50a12b974/ts/Maps/Projection.ts#L158).

Reply:
Thanks for good feedback!

Reply:
On the master branch, we've got logic to get a box of the rotated label: https://jsfiddle.net/BlackLabel/uLt7rn95/ It is positioned to (0,0) because that was needed in other methods in HC.

Reply:
I've got a PR ready for it here: https://github.com/highcharts/highcharts/pull/20641 . Can't see any problems with it now... https://jsfiddle.net/highcharts/wng69bsq/